src/settings/components/saveButton.tsx

```diff
@@ -55,9 +60,14 @@ export default function SaveButton({
       dispatch(setNumberOfMealsPerDay(Number(numberMeals)));
       dispatch(setNumberOfMinutesToFirstMeal(minuteToFirstMeal));
     });
-    myLocalStorage.saveSettings(settings);
     firstEntry.markAsUsed();
+    goToWindowWithBtn();
   };
+  const settings = useSelector(selectSettings);
+
+  useEffect(() => {
+    myLocalStorage.saveSettings(settings);
+  });
```

Comment:
remember that if you leave the dependencies array (2nd arg to useEffect) blank, then your effect will trigger on every render of a react component, it's almost never needed and may lead to unexpected bugs

Reply:
the updated code doesn't use  the useEffect